### PR TITLE
Feature 9 - ToggleSelector widget

### PR DESCRIPTION
**Pull request by Franek75**
# PR Details

ToggleSelector widget added

## Description

* ToggleSelector, as it has been recently reshaped for the project
* SelectOptionById and SelectOptionByText methods are now included

NOTE: including package-lock.json was not intentional

## Related Issue

[#9 Add ToggleSelector widget and implement ElementFinder's byCSSContainingText() method](https://github.com/systelab/systelab-components-wdio-test/issues/9) 

## Motivation and Context

Make the widget from the project available to everybody else

## How Has This Been Tested

By using it in the project.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation 
- [ ] I have updated the documentation accordingly (README.md for each widget)
- [ ] I have added tests to cover my changes (at least 1 spec for each widget with the same coverage as the master branch)
- [ ] All tests (new and existing) passed
- [ ] A new branch needs to be created from master to evolve previous versions
- [x] Increase version in package.json following [Semantic Versioning](https://semver.org/)
- [x] Add the issue into the right [project](https://github.com/systelab/systelab-components-wdio-test/projects) with the proper status (In progress)

### CODE REVIEW

File: src/widgets/toggle-selector.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+export class ToggleSelector extends Widget {
+
+    public async selectOptionById(optionId: string) {
+        await this.byId(optionId).click();
+    }
+
+    public async getSelectedOption(): Promise<string> {
+        return this.byClassName('slab-option slab-selected ng-star-inserted').getText();
```
Comment (joaquimvila) [inline src/widgets/toggle-selector.ts:16]: Do we really need to include "ng-star-inserted" in the class selector? Can't this be done with an expression that better facilitates identifying the element with the option text?

File: src/widgets/toggle-selector.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { Widget } from './widget';
+import { ElementArrayFinder } from '../wdio';
+
+export interface ToggleSelectorOption {
+    id: string;
+    caption: string;
+}
+
+export class ToggleSelector extends Widget {
+
```
Comment (joaquimvila) [inline src/widgets/toggle-selector.ts:10]: What about a method to select an option by text? 
If so, using this widget wouldn't require using the interface defined above where we are exposing an internal detail (id) of the toggle selector implementation.

File: src/widgets/toggle-selector.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        return this.byClassName('slab-option slab-selected ng-star-inserted').getText();
+    }
+
+    public async getOptions(): Promise<ToggleSelectorOption[]> {
+        const optionArray: ToggleSelectorOption[] = [];
+        const options: ElementArrayFinder = this.allByCSS("div[class*='slab-option']");
+        const numberOfOptions: number = await options.count();
+        for (let i = 0; i < numberOfOptions; i++) {
+            // We need to remove the first 2 chars because id's are being created always with 'id' prefixing the string
+            const id: string = (await options.get(i).getAttribute('id')).slice(2);
```
Comment (joaquimvila) [inline src/widgets/toggle-selector.ts:25]: Why not identifying the available options by index (0-based)? If so, the return type of this method could be just an array of strings and the index is implicit by indicating the position of the element on the array.

Comment (joaquimvila) [reply src/widgets/toggle-selector.ts:25]: If going for that approach, it would be great to have a method named "selectOptionByIndex"

**Review by joaquimvila — CHANGES_REQUESTED** [review]
Good job @Franek75 with this PR! However, I think that now it's the right time to harmonize the  interface for this new widget object with the one of other "option selection" widgets. It can be done right now because the impact on the projects would be minimal. I see this as a combobox, so from a test perspective both interfaces should be equivalent.

File: src/widgets/toggle-selector.ts (hunk 2 above)
Comment (Franek75) [reply src/widgets/toggle-selector.ts:10]: It was initially intended, but I deferred it (see description)

File: src/widgets/toggle-selector.ts (hunk 1 above)
Comment (Franek75) [reply src/widgets/toggle-selector.ts:16]: I am not sure. At that moment it was the only/quick way I found. It could be improved

Comment (Franek75) [reply src/widgets/toggle-selector.ts]: I have uploaded ToggleSelector WO's code as it was recently reshaped by @joaquimvila for the project.
Thus a brand new review is needed

File: src/widgets/toggle-selector.ts (hunk 3 above)
Comment (Franek75) [reply src/widgets/toggle-selector.ts]: I have uploaded ToggleSelector WO's code as it was recently reshaped by @joaquimvila for the project.
Thus a brand new review is needed

**Review by joaquimvila — APPROVED** [review]
Good job @Franek75! Thanks for sharing this with the rest of the company

**Pull request merged**